horace_core/admin/upgrade_file_format.m

```diff
@@ -46,20 +41,24 @@
 for i=1:n_inputs
     if is_sqw(i)
         ld = sqw_formats_factory.instance().get_loader(filenames{i});
-        if isa(ld,'faccess_sqw_v4') && upgrade_ranges %
-            if nargout > 1
-                sqw_list{i} = finalize_alignment(ld);
-            else
-                finalize_alignment(ld);   % Will do nothing if the file is not aligned
-            end
+        pref_acc = sqw_formats_factory.instance().get_pref_access('sqw');
+        if isa(ld,class(pref_acc))
+            ld_new = ld;
         else
-            ld_new = ld.upgrade_file_format(upgrade_arg{:});
+            ld_new  = ld.upgrade_file_format();
+            ld.delete();
+        end
+
+        if upgrade_ranges %
             if nargout > 0
-                sqw_list{i} = sqw(ld_new,'file_backed',true);
+                sqw_list{i} = finalize_alignment(ld_new);
+            else
+                finalize_alignment(ld_new);   % Will do nothing if the file is not aligned && ranges are valid
             end
-            ld_new.delete();
+        elseif nargout>0
+            sqw_list{i} = sqw(ld_new,'file_backed',true);
         end
```

Comment:
Would prefer like elements grouped together
```suggestion
        if nargout > 0
            if upgrade_ranges %
                sqw_list{i} = finalize_alignment(ld_new);
            else
                sqw_list{i} = sqw(ld_new,'file_backed',true);
            end
        else
            finalize_alignment(ld_new);   % Will do nothing if the file is not aligned && ranges are valid
        end
```

Reply:
I am very reluctant to change order of operations here on a whim. The operations have costs there and in suggested case you will read metadata regardless of your intentions on aligning them. This change needs much more thinking than I am ready to spend on it